src/common/load-plugins.js

```diff
@@ -5,18 +5,43 @@ const partition = require("lodash/partition");
 const fs = require("fs");
 const globby = require("globby");
 const path = require("path");
+const pkgDir = require("pkg-dir").sync;
 const thirdParty = require("./third-party");
 const internalPlugins = require("./internal-plugins");
 const mem = require("mem");
 const resolve = require("./resolve");
 
 const memoizedLoad = mem(load, { cacheKey: JSON.stringify });
 const memoizedSearch = mem(findPluginsInNodeModules);
+const memoizedPluginAutoSearchDirs = mem(pluginAutoSearchDirs);
 const clearCache = () => {
   mem.clear(memoizedLoad);
   mem.clear(memoizedSearch);
+  mem.clear(memoizedPluginAutoSearchDirs);
 };
 
+// unless pluginSearchDirs are provided, auto-load plugins from node_modules that are parent to Prettier
+function pluginAutoSearchDirs() {
+  const pluginSearchDirs = [];
+  const packageRoot = pkgDir(__dirname);
+
+  const autoLoadDir = thirdParty.findParentDir(
+    path.join(packageRoot, ".."),
+    "node_modules"
+  );
+
+  if (autoLoadDir) {
+    pluginSearchDirs.push(autoLoadDir);
+  }
+
+  // We may have plugins in `prettier/prettier` codebase for test
+  if (isDirectory(path.join(packageRoot, "node_modules"))) {
+    pluginSearchDirs.push(packageRoot);
+  }
+
+  return pluginSearchDirs;
+}
```

Comment:
What about yarn pnp?

Reply:
We don't support it.


Possible solution `--plugin=@prettier/plugin-pug`

Reply:
I think will be great to support them in the future

Reply:
@evilebottnawi I'm interested in your thoughts on https://github.com/prettier/prettier/issues/8474#issuecomment-639013515.

Reply:
You shouldn't have to reinvent the node resolution, why not use `require.resolve` to find the plugin from the project that depends on prettier / whatever invokes it?

Reply:
Because we don't know the plugin name yet.

Reply:
Oh, I assumed you loaded the name from `package.json` which would be the sensible thing to do

Reply:
Currently, we search `node_modules`, so some plugins might be "indirect" dependencies.

Reply:
Imaging this, you have `@merceyz/prettier` package, include `config` and `plugins`, you may don't want list all plugins in every project.

Reply:
Then I would assume `@merceyz/prettier` declares the plugins name/location in a config and it's loaded from there, basically the same way babel handles plugins/presets

Reply:
Yep, that's my proposal https://github.com/prettier/prettier/issues/8474#issuecomment-661543282, but we haven't support it.